api/envoy/extensions/transport_sockets/tls/v3/tls.proto

```diff
@@ -96,10 +99,24 @@ message DownstreamTlsContext {
 }
 
 // TLS context shared by both client and server TLS contexts.
-// [#next-free-field: 9]
+// [#next-free-field: 11]
 message CommonTlsContext {
   option (udpa.annotations.versioning).previous_message_type = "envoy.api.v2.auth.CommonTlsContext";
 
+  // Config for Certificate provider to get certificates. This provider should allow certificates to be
+  // fetched/refreshed over the network "asynchronously" with respect to the TLS handshake.
+  message CertificateProvider {
+    // certificate name: can be used to request a local cert vs a root cert
+    string name = 1;
+
+    // Provider specific config: contains a name and an Any struct to pass a typed config.
```

Comment:
The "contains name/Any" is redundant,  this is implied by the type and is a standard feature in the API.